distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -698,21 +698,26 @@ type Table
     evaluate_expression : Text -> Problem_Behavior -> Column ! No_Such_Column | Invalid_Value_Type | Expression_Error
     evaluate_expression self expression on_problems=Report_Warning =
         get_column name = self.at name
-        type_mapping = self.connection.dialect.get_type_mapping
-        make_constant value =
-            argument_value_type = Value_Type_Helpers.find_argument_type value
-            sql_type = case argument_value_type of
-                Nothing -> SQL_Type.null
-                _ -> type_mapping.value_type_to_sql argument_value_type Problem_Behavior.Ignore
-            expr = SQL_Expression.Constant value
-            new_type_ref = SQL_Type_Reference.from_constant sql_type
-            Column.Value ("Constant_" + UUID.randomUUID.to_text) self.connection new_type_ref expr self.context
-        new_column = Expression.evaluate expression get_column make_constant "Standard.Database.Data.Column" "Column" Column.var_args_functions
+        new_column = Expression.evaluate expression get_column self.make_constant_column "Standard.Database.Data.Column" "Column" Column.var_args_functions
         problems = Warning.get_all new_column . map .value
         result = new_column.rename (self.connection.dialect.get_naming_helpers.sanitize_name expression)
         on_problems.attach_problems_before problems <|
             Warning.set result []
 
+    ## PRIVATE
+
+       Create a constant column from a value.
+    make_constant_column : Any -> Column
+    make_constant_column self value =
```

Comment:
I'd add
```
if is_column value then Error.throw (Illegal_Argument.Error "A constant value may only be created from a scalar, not a Column") else
```

Reply:
Mainly because I'd think this would usually mean an error (that where a constant was expected, a column was provided - likely a `zip`-like operation was needed instead)

Reply:
Done.